tests/drivers/regulator/voltage/boards/frdm_mcxn947_mcxn947_cpu0.overlay

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright 2024 NXP
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/dt-bindings/adc/adc.h>
+#include <zephyr/dt-bindings/adc/mcux-lpadc.h>
+#include <zephyr/dt-bindings/regulator/nxp_vref.h>
+
+/* To do this test, connect LPADC0 channel 2A(J8 pin 28) to VREF_OUT (TP1) */
+
+/ {
+	resources: resources {
+		compatible = "test-regulator-voltage";
+		regulators = <&vref>;
+		tolerance-microvolt = <10000>;
+		set-read-delay-ms = <1>;
+		adc-avg-count = <10>;
+		io-channels = <&lpadc0 0>;
+		min-microvolt = <1000000>;
+		max-microvolt = <2100000>;
+	};
+};
+
+&vref {
+	regulator-initial-mode = <NXP_VREF_MODE_INTERNAL_REGULATOR>;
+};
+
+&lpadc0 {
+	#address-cells = <1>;
+	#size-cells = <0>;
+
+	voltage-ref= <2>;
```

Comment:
Just a nit- might be worth mentioning here in a comment that this switches the ADC source to no longer use the VREF as a source.

Also, won't the LPADC driver still try to set the regulator's voltage in this case? Perhaps we need to delete this property from the node for this overlay with the `/delete-node/` property

Reply:
the LPADC will not set the regulator's voltage, in this case, the **zephyr,reference** property is set to **ADC_REF_EXTERNAL0**, which means the regulator will not be stetted in the LPADC driver.